src/plugins/average.rs

```diff
@@ -0,0 +1,106 @@
+    fn new() -> Average {
+        Average { total: None, count: 1 }
+    }
+
+    fn average(&mut self, value: Tagged<Value>) -> Result<(), ShellError> {
+        match value.item() {
+            Value::Primitive(Primitive::Nothing) => Ok(()),
+            Value::Primitive(Primitive::Int(i)) => {
+                match &self.total {
+                    Some(Tagged {
+                        item: Value::Primitive(Primitive::Int(j)),
+                        tag,
+                    }) => {
+                        self.total = Some(Value::int(i + j).tagged(tag));
+                        self.count = self.count + 1;
+                        Ok(())
+                    }
+                    None => {
+                        self.total = Some(value.clone());
+                        Ok(())
+                    }
+                    _ => Err(ShellError::string(format!(
+                        "Could not calculate average of non-integer or unrelated types"
+                    ))),
+                }
+            }
+            Value::Primitive(Primitive::Bytes(b)) => {
+                match self.total {
+                    Some(Tagged {
+                        item: Value::Primitive(Primitive::Bytes(j)),
+                        tag,
+                    }) => {
+                        self.total = Some(Value::int(b + j).tagged(tag));
+                        self.count = self.count + 1;
+                        Ok(())
+                    }
+                    None => {
+                        self.total = Some(value);
+                        Ok(())
+                    }
+                    _ => Err(ShellError::string(format!(
+                        "Could not calculate average of non-integer or unrelated types"
+                    ))),
+                }
+            }
+            x => Err(ShellError::string(format!(
+                "Unrecognized type in stream: {:?}",
+                x
+            ))),
+        }
+
+    }
+}
+
+impl Plugin for Average {
+    fn config(&mut self) -> Result<Signature, ShellError> {
+        Ok(Signature::build("average")
+            .desc("Compute the average of a column of numerical values.")
+            .filter())
+    }
+
+    fn begin_filter(&mut self, _: CallInfo) -> Result<Vec<ReturnValue>, ShellError> {
+        Ok(vec![])
+    }
+
+    fn filter(&mut self, input: Tagged<Value>) -> Result<Vec<ReturnValue>, ShellError> {
+        self.average(input)?;
+        Ok(vec![])
+    }
+
+    fn end_filter(&mut self) -> Result<Vec<ReturnValue>, ShellError> {
+        match self.total {
+            None => Ok(vec![]),
+            Some(ref v) =>  {
+                match v.item() {
+                    Value::Primitive(Primitive::Int(i)) => {
+                        let total: u64 = i.tagged(v.tag).coerce_into("converting for average")?;
+                        let avg = total as f64 / self.count as f64;
+                        let decimal_value: Value=  Primitive::from(avg).into();
```

Comment:
Something you'll probably also want to do (that I *think* the CI will also check for) is running your source against `rustfmt`.

You can install it using `rustup component add rustfmt`. From there you can run it, or have editors like vscode run it for you. This will keep the formatting of your source file uniform.

Reply:
oh, whoops! I knew I forgot to do something before submitting. I should make a vim or git hook for this.